hledger/test/query-tag.test

```diff
@@ -214,3 +214,23 @@ $ hledger -f- bal -N tag:type=a
 # 21.
 $ hledger -f- reg -w80 tag:type=a
 2022-01-01                      (a:aa)                           1             1
+
+# 22. Postings can override the tags of their parents
+<
+2022-11-17 Aldi
+    ; concerns: me
+    Assets               -30 €
+    Costs:Food            20 €
+    Loaned                10 €  ; concerns: you
+
+$ hledger -f- reg tag:concerns=me --pivot=concerns
+2022-11-17 Aldi                 me                           -30 €         -30 €
+                                me                            20 €         -10 €
+
+# 23. Accounts can override the tags of their parents
+<
+account a    ; atag:A
+account a:b  ; atag:B
+
+$ hledger -f- accounts tag:atag=a
+a
```

Comment:
There could be more tests for the combination of transaction/posting and account tags:

```suggestion
a

# 24. Transactions can override the tags of used accounts
<
account a    ; atag:A
account a:b  ; atag:B

2023-01-01 Test
    ; atag:C
    (a:b)  10 €

$ hledger -f- bal --pivot=atag
                10 €  C
--------------------
                10 €

# 25. Postings can override the tags of their accounts
<
account a    ; atag:A
account a:b  ; atag:B

2023-01-01 Test
    ; atag:C
    (a:b)  10 €  ; atag:D

$ hledger -f- bal --pivot=atag
                10 €  D
--------------------
                10 €
```

Reply:
I've added your tests as suggested and I'm thinking of more tests that should be relevant to this ordering with multi-valued tags. Thank you very much for taking the time to look into some tests!